src/mapblock.cpp

```diff
@@ -381,63 +442,19 @@ void MapBlock::serialize(std::ostream &os, u8 version, bool disk, int compressio
 	/*
 		Bulk node data
 	*/
-	NameIdMapping nimap;
-	if(disk)
-	{
-		MapNode *tmp_nodes = new MapNode[nodecount];
-		for(u32 i=0; i<nodecount; i++)
-			tmp_nodes[i] = data[i];
-		getBlockNodeIdMapping(&nimap, tmp_nodes, m_gamedef->ndef());
-
-		u8 content_width = 2;
-		u8 params_width = 2;
-		writeU8(os, content_width);
-		writeU8(os, params_width);
-		MapNode::serializeBulk(os, version, tmp_nodes, nodecount,
-				content_width, params_width, compression_level);
-		delete[] tmp_nodes;
-	}
-	else
-	{
-		u8 content_width = 2;
-		u8 params_width = 2;
-		writeU8(os, content_width);
-		writeU8(os, params_width);
-		MapNode::serializeBulk(os, version, data, nodecount,
-				content_width, params_width, compression_level);
-	}
+	u8 content_width = 2;
+	u8 params_width = 2;
+	writeU8(os, content_width);
+	writeU8(os, params_width);
+	MapNode::serializeBulk(os, version, data, nodecount,
+			       content_width, params_width, compression_level);
 
 	/*
 		Node metadata
 	*/
 	std::ostringstream oss(std::ios_base::binary);
 	m_node_metadata.serialize(oss, version, disk);
 	compressZlib(oss.str(), os, compression_level);
-
-	/*
-		Data that goes to disk, but not the network
-	*/
-	if(disk)
-	{
-		if(version <= 24){
```

Comment:
To reviewers: This part of the function now only handles network serialization. These deleted lines were moved to ` MapBlockDiskSer::compress`.